src/Payments.sol

```diff
@@ -47,6 +47,20 @@ contract Payments is
 
     uint256 public constant PAYMENT_FEE_BPS = 10; //(0.1 % fee)
 
+    // Events
+    event AccountLockupSettled(address indexed token, address indexed owner, uint256 lockupCurrent, uint256 lockupRate, uint256 lockupLastSettledAt);
+    event OperatorApprovalSet(address indexed token, address indexed client, address indexed operator, bool approved);
+
+    event RailCreated(uint256 indexed railId);
+    event RailTerminated(uint256 indexed railId, uint256 endEpoch);
+    event RailLockupModified(uint256 indexed railId, uint256 newLockupPeriod, uint256 newLockupFixed);
+    event RailPaymentRateModified(uint256 indexed railId, uint256 oldRate, uint256 newRate);
+    event RailSettled(uint256 indexed railId, uint256 totalSettledAmount, uint256 totalNetPayeeAmount, uint256 paymentFee, uint256 operatorCommission, uint256 settledUpTo);
```

Comment:
@silent-cipher Can we also add an event for `finalizeTerminatedRail` ? Take a look at the function to see how it is used/called. An event for that would be nice to have.